src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -0,0 +1,481 @@
+import _drop from 'lodash/drop';
+import _findIndex from 'lodash/findIndex';
+import _get from 'lodash/get';
+import _isEmpty from 'lodash/isEmpty';
+import _isNil from 'lodash/isNil';
+import _isObject from 'lodash/isObject';
+import _map from 'lodash/map';
+import ModeController from '../ModeControl/ModeController';
+import LegModel from './LegModel';
+import { routeStringFormatHelper } from '../../navigationLibrary/Route/routeStringFormatHelper';
+import {
+    MCP_MODE,
+    MCP_MODE_NAME,
+    MCP_PROPERTY_MAP
+} from '../ModeControl/modeControlConstants';
+
+/**
+ *
+ *
+ * This class should always be instantiated from an `AircraftInstanceModel` and
+ * always instantiated with some form of a `spawnPatternModel`.
+ *
+ * @class Fms
+ */
+export default class Fms {
+    /**
+     * @constructor
+     * @param aircraftInitProps {object}
+     * @param initialRunwayAssignment {string}
+     * @param navigationLibrary {NavigationLibrary}
+     */
+    constructor(aircraftInitProps, initialRunwayAssignment, typeDefinitionModel, navigationLibrary) {
+        if (!_isObject(aircraftInitProps) || _isEmpty(aircraftInitProps)) {
+            throw new TypeError('Invalid aircraftInitProps passed to Fms');
+        }
+
+        /**
+         * Instance of the `NavigationLibrary`
+         *
+         * provides access to the aiport SIDs, STARs and Fixes via collection objects and fascade methods.
+         * used for route building
+         *
+         * @property _navigationLibrary
+         * @type {NavigationLibrary}
+         * @private
+         */
+        this._navigationLibrary = navigationLibrary;
+
+        /**
+         * Keeps track of current Altitude, Heading and Speed modes and values.
+         *
+         * @property _modeController
+         * @type {ModeController}
+         */
+        this._modeController = new ModeController(typeDefinitionModel);
+
+        /**
+         *
+         *
+         * @property _aircraftTypeDefinition
+         * @type {AircraftTypeDefinitionModel}
+         * @private
+         */
+        this._aircraftTypeDefinition = typeDefinitionModel;
+
+        /**
+         * Name of the initial runway assigned for takeoff/landing.
+         *
+         * This value is likely to change as an aircraft moves through the airspace.
+         *
+         * @property _runway
+         * @type {string}
+         * @private
+         */
+        this._runwayName = initialRunwayAssignment;
```

Comment:
property name doesn't match docblock.

We could store runway assignments in the fms altogether I suppose. They shouldn't exist _both_ in the fms and the aircraft, so we could move all of that to here to centralize things a bit.